src/BuildIntegration/Microsoft.DotNet.ILCompiler.targets

```diff
@@ -0,0 +1,44 @@
+<Project 
+  xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <DeployAppHost>false</DeployAppHost>
+    <_ILCompilerPackageRootDir>$(MSBuildThisFileDirectory)../</_ILCompilerPackageRootDir>
+    <IlcBuildIntegrationTargetsDir>$(_ILCompilerPackageRootDir)build/</IlcBuildIntegrationTargetsDir>
+   
+    <!-- Workaround for lack of current host OS detection - https://github.com/Microsoft/msbuild/issues/539 -->
+    <TargetOS Condition="'$(TargetOS)' == '' and '$(OS)' == 'Unix' and Exists('/Applications')">OSX</TargetOS>
+    <!-- The correct OS detection code. Uncomment once CI machines are upgraded to the latest version of MSBuild -->
+    <!-- <TargetOS Condition="'$([MSBuild]::IsOSPlatform(OSX))' == 'true'">OSX</TargetOS>  -->
+    <TargetOS Condition="'$(TargetOS)' == ''">$(OS)</TargetOS>
```

Comment:
This is ok for now. 

Eventually, we should just use the target that is passed on the `dotnet publish` command line via `-r` switch.